### DISCUSSION TITLE
synctex: highlight position in source upon rebuilding

### DISCUSSION THREAD

**Discussion by tdegeus** [James-Yu/LaTeX-Workshop#3385] [Q&A]
Thanks for the great extension!!

I would find it very intuitive if, upon saving (if rebuilding), the position in the PDF would be highlighted. I.e. to have `cmd + s` be the same as what is know `cmd + s` and then `cmd + option + j`. 

### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you can not see the requested feature. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before requesting a feature.

You visited the wiki?: Yes

### If your requested feature is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: Yes

## Is your requested feature  related to a problem? Please describe. [Required]

-

## To Reproduce  [Required]

-

## Logs [Required]

-
### LaTeX Workshop Output [Required]

-

### Developer Tools Console [Required]

-

## Screenshots

-

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: macOS
 - VS Code version: 1.69.2 (Universal)
 - Extension version: 8.27.2
 - TeX distribution version: macTeX 2021

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

### Additional context

-

**Comment by James-Yu** [comment]
Not likely to be implemented. It's quite an edge request unfortunately.

edit: you may find possible alternatives using the built-in vscode task system.

**Comment by yarakos95** [comment]
@tdegeus

I apologise in advance. This is an idea, but it does not work well.

By using [multi-command][mp_mc], you can use multiple commands as one sequence. By configuring as follows, `latex-workshop.synctex` will be executed after `latex-workshop.build` is executed.

[mp_mc]: https://marketplace.visualstudio.com/items?itemName=ryuta46.multi-command

```jsonc
// settings.json
  "multiCommand.commands": [
    {
      "command": "multiCommand.latex.buildAndSyncTeXJump",
      "sequence": [
        {
          "command": "latex-workshop.build",
          "onSuccess": [
            "latex-workshop.synctex",
          ]
        }
      ],
      "label": "buildAndSyncTeXJump",
      "description": "The position in PDF would be highlighted after TeX file has been built.",
      "languages": ["latex"]
    }
  ],
```

This assumes that it will be run with internal PDF viewer open. If you configure keybindings.json as follows, the above sequence will be executed at save time.

```jsonc
// keybindings.json
    {
        "key": "ctrl+s",
        "command": "extension.multiCommand.execute",
        "args": { "command": "multiCommand.latex.buildAndSyncTeXJump" },
        "when": "editorTextFocus && editorLangId == latex"
    },
```

However, this does not work properly. The reason why this configuration doesn't work as expected seems to come from the following issue.

> <https://github.com/ryuta46/vscode-multi-command/issues/23>

If this issue is resolved, you should get the expected behavior with the above configuration.
